monitor/services/src/main/java/org/datacleaner/monitor/server/components/ComponentHandler.java

```diff
@@ -448,4 +487,105 @@ public void onValues(Object[] values) {
         }
     }
 
+    private class ComponentHandlerAnalysisJob implements AnalysisJob {
+
+        @Override
+        public AnalysisJobMetadata getMetadata() {
+            return null;
+        }
+
+        @Override
+        public Datastore getDatastore() {
+            return null;
+        }
+
+        @Override
+        public List<InputColumn<?>> getSourceColumns() {
+            return inputColumnsList;
+        }
+
+        @Override
+        public synchronized List<TransformerJob> getTransformerJobs() {
+            return Collections.singletonList((TransformerJob)new ComponentHandlerTransformerJob());
+        }
+
+        @Override
+        public List<FilterJob> getFilterJobs() {
+            return Collections.emptyList();
+        }
+
+        @Override
+        public List<AnalyzerJob> getAnalyzerJobs() {
+            return Collections.emptyList();
+        }
+    }
+
+    class ComponentHandlerTransformerJob implements TransformerJob {
+
+        @Override
+        public TransformerDescriptor<?> getDescriptor() {
+            return (TransformerDescriptor<?>)descriptor;
+        }
+
+        @Override
+        public String getName() {
+            return descriptor.getDisplayName();
+        }
+
+        @Override
+        public Map<String, String> getMetadataProperties() {
+            return Collections.emptyMap();
+        }
+
+        @Override
+        public org.datacleaner.job.ComponentConfiguration getConfiguration() {
+            return config;
+        }
+
+        @Override
+        public ComponentRequirement getComponentRequirement() {
+            return null;
+        }
+
+        @Override
+        public InputColumn<?>[] getInput() {
+            return configuredInputColumns;
+        }
+
+        @Override
+        public InputColumn<?>[] getOutput() {
+            // TODO
+            return new InputColumn<?>[0];
+        }
+
+        @Override
+        public OutputDataStreamJob[] getOutputDataStreamJobs() {
+            return new OutputDataStreamJob[0];
+        }
+    }
+
+    private class ComponentHandlerInjectionManager implements InjectionManager {
+        InjectionManager delegate;
+        ComponentDescriptor componentDescriptor;
+        Component component;
+
+        ComponentHandlerInjectionManager(InjectionManager delegate, ComponentDescriptor<?> componentDescriptor, Component component) {
+            this.delegate = delegate;
+            this.component = component;
+            this.componentDescriptor = componentDescriptor;
+        }
+
+        public <E> E getInstance(InjectionPoint<E> injectionPoint) {
+            E obj;
+            final Class<E> baseType = injectionPoint.getBaseType();
+            if (baseType == OutputRowCollector.class) {
+                obj = (E) new ThreadLocalOutputRowCollector();
+            } else if(baseType == ComponentContext.class) {
+                obj = (E) componentContext;
+            } else {
```

Comment:
PlaceholderAnalysisJob unfortunately cannot be reused nor refactored to be usable here. Here in ComponentHandler we have not "full" job - there is no real datastore, etc... So in this case I implemented some lighter AnalysisJob, using data directly from ComponentHandler...